src/mlpack/bindings/julia/strip_type.hpp

```diff
@@ -0,0 +1,57 @@
+/**
+ * @filename strip_type.hpp
+ * @author Ryan Curtin
+ *
+ * Given a C++ type name, turn it into something that has no special characters
+ * that can simply be printed.
+ */
+#ifndef MLPACK_BINDINGS_JULIA_STRIP_TYPE_HPP
+#define MLPACK_BINDINGS_JULIA_STRIP_TYPE_HPP
+
+namespace mlpack {
+namespace bindings {
+namespace julia {
+
+/**
+ * Given a C++ type name, turn it into something that has no special characters
+ * that can simply be printed.  This is similar to but not identical to
+ * mlpack::bindings::python::StripType().
+ *
+ * @param cppType C++ type as a string.
+ * @return Stripped type with no special characters.
+ */
+inline std::string StripType(std::string cppType)
+{
+  // Basically what we need to do is strip any '<' (template bits) from the
+  // type.
+  if (cppType.find("<") != std::string::npos)
+  {
+    // Are there any template parameters?  Or is it the default?
+    const size_t loc = cppType.find("<>");
+    if (loc != std::string::npos)
+    {
+      // Convert it from "<>".
+      cppType.replace(loc, 2, "");
+    }
+    else
+    {
+      // Let's just replace the '<' and '>' with a valid '_' character.
+      while (cppType.find("<") != std::string::npos)
```

Comment:
Not sure I see the reason to use ``find`` on the string first, as ``cppType.replace(cppType.begin(), cppType.end(), "<", "_");`` should do the same?

Reply:
Which overload of `std::string::replace()` is that?  When I read the documentation at http://www.cplusplus.com/reference/string/string/replace/ I can't figure out which one it is.  Maybe I overlooked it?

Reply:
You are right, I was talking about: ``std::replace``, so ``std::replace(cppType.begin(), cppType.end(), "<", "_");``.

Reply:
Ah, got it; I refactored to use `std::replace()` in 74620db49.  Thanks for the suggestion, it is a lot nicer. :+1: